tools/kagami/src/main.rs

```diff
@@ -705,6 +705,10 @@ mod docs {
 }
 
 mod validator {
```

Comment:
I'd suggest breaking up `main` into smaller modules, given that we have them conveniently defined, just use `rust-analyzer` to inline them.